lang/shared/src/main/scala/com/wavesplatform/lang/v1/evaluator/ctx/impl/waves/WavesContext.scala

```diff
@@ -18,36 +19,88 @@ object WavesContext {
   def build(env: Environment): CTX = {
     val environmentFunctions = new EnvironmentFunctions(env)
 
-    def getdataF(name: String, internalName: Short, dataType: DataType) =
-      PredefFunction(name, 100, OPTION(dataType.innerType), List(("address", addressType.typeRef), ("key", STRING)), internalName) {
+    def getdataF(name: String, internalName: Short, dataType: DataType): BaseFunction =
+      NativeFunction(name, 100, internalName, OPTION(dataType.innerType), "address" -> addressType.typeRef, "key" -> STRING) {
         case (addr: CaseObj) :: (k: String) :: Nil => environmentFunctions.getData(addr, k, dataType)
         case _                                     => ???
       }
 
-    val getLongF: PredefFunction      = getdataF("getLong", DATA_LONG, DataType.Long)
-    val getBooleanF: PredefFunction   = getdataF("getBoolean", DATA_BOOLEAN, DataType.Boolean)
-    val getByteArrayF: PredefFunction = getdataF("getByteArray", DATA_BYTES, DataType.ByteArray)
-    val getStringF: PredefFunction    = getdataF("getString", DATA_STRING, DataType.String)
+    val getLongF: BaseFunction      = getdataF("getLong", DATA_LONG, DataType.Long)
+    val getBooleanF: BaseFunction   = getdataF("getBoolean", DATA_BOOLEAN, DataType.Boolean)
+    val getByteArrayF: BaseFunction = getdataF("getByteArray", DATA_BYTES, DataType.ByteArray)
+    val getStringF: BaseFunction    = getdataF("getString", DATA_STRING, DataType.String)
+
+    def secureHashExpr(xs: EXPR): EXPR = FUNCTION_CALL(
+      FunctionHeader.Native(KECCAK256),
+      List(
+        FUNCTION_CALL(
+          FunctionHeader.Native(BLAKE256),
+          List(xs)
+        )
+      )
+    )
 
-    val addressFromPublicKeyF: PredefFunction =
-      PredefFunction("addressFromPublicKey", 100, addressType.typeRef, List(("publicKey", BYTEVECTOR)), ADDRESSFROMPUBKEY) {
-        case (pk: ByteVector) :: Nil =>
-          val r = environmentFunctions.addressFromPublicKey(pk)
-          Right(CaseObj(addressType.typeRef, Map("bytes" -> r)))
-        case _ => ???
-      }
+    val addressFromBytesF = NativeFunction("addressFromBytes", 1, ADDRESSFROMBYTES, addressType.typeRef, "bytes" -> BYTEVECTOR) {
+      case (bytes: ByteVector) :: Nil => Right(CaseObj(addressType.typeRef, Map("bytes" -> bytes)))
+      case _                          => ???
+    }
+
+    val addressFromPublicKeyF: BaseFunction = UserFunction("addressFromPublicKey", 100, addressType.typeRef, "publicKey" -> BYTEVECTOR) {
+      case pk :: Nil =>
+        Right(
+          FUNCTION_CALL(
+            FunctionHeader.Native(ADDRESSFROMBYTES),
+            List(
+              BLOCK(
+                LET(
+                  "@afpk_withoutChecksum",
+                  FUNCTION_CALL(
+                    FunctionHeader.Native(SUM_BYTES),
+                    List(
+                      CONST_BYTEVECTOR(ByteVector(EnvironmentFunctions.AddressVersion, env.networkByte)),
```

Comment:
I'll reference `EnvironmentFunctions.AddressVersion` and other constants in RIDE's variables in a next PR.